src/ott/geometry/geodesic.py

```diff
@@ -0,0 +1,282 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Dict, Optional, Sequence, Tuple
+
+import jax
+import jax.experimental.sparse as jesp
+import jax.numpy as jnp
+import numpy as np
+from scipy.special import ive
+
+from ott import utils
+from ott.geometry import geometry
+from ott.math import utils as mu
+from ott.types import Array_g
+
+__all__ = ["Geodesic"]
+
+
+@jax.tree_util.register_pytree_node_class
+class Geodesic(geometry.Geometry):
+  r"""Graph distance approximation using heat kernel :cite:`huguet:2022`.
+
+  Approximates the heat kernel using Chebyshev polynomials of the
+  first kind of max order ``order``, which for small ``t`` approximates the
+  geodesic exponential kernel :math:`e^{\frac{-d(x, y)^2}{t}}`.
+
+  Args:
+    scaled_laplacian: The Laplacian scaled by the largest eigenvalue.
+    eigval: Largest eigenvalue of the Laplacian.
+    chebyshev_coeffs: Coefficients of the Chebyshev polynomials.
+    t: Time parameter for heat kernel.
+    order: Max order of Chebyshev polynomial.
+    kwargs: Keyword arguments for :class:`~ott.geometry.geometry.Geometry`.
+  """
+
+  def __init__(
+      self,
+      scaled_laplacian: Array_g,
+      eigval: jnp.ndarray,
+      chebyshev_coeffs: jnp.ndarray,
+      t: float = 1e-3,
+      order: int = 100,
+      **kwargs: Any
+  ):
+    super().__init__(epsilon=1., **kwargs)
+    self.scaled_laplacian = scaled_laplacian
+    self.eigval = eigval
+    self.chebyshev_coeffs = chebyshev_coeffs
+    self.t = t
+    self.order = order
```

Comment:
I only removed `self.order`, because `self.t` is used for the attribute `cost_matrix`, and `self.eigval`, `self.chebyshev_coeffs` are used in`expm_multiply`. I renamed `phi` to `eigval` in `expm_multiply` to make it clearer.